### PR TITLE
Fix error where any page with the word "submit" in its name would cause the submission to submit

### PR DESCRIPTION
**Pull request by bseeger**
I just discovered that if a page has the word "submit" in the name of the page, it will cause the submission to get submitted. 

Thankfully the only examples of where this happens in state projects is where we do want the data marked as submitted. 

Tests are updated as well.

### CODE REVIEW

File: src/main/java/formflow/library/ScreenController.java [hunk 1]
```diff
@@ -197,7 +197,7 @@ ModelAndView postScreen(
     handleAddressValidation(submission, formSubmission);
 
     // handle submit actions, if requested
-    if (request.getRequestURI().toLowerCase().contains("submit")) {
+    if (request.getRequestURI().toLowerCase().endsWith("/submit")) {
```
Comment (bseeger) [inline src/main/java/formflow/library/ScreenController.java:200]: This still doesn't feel good enough, but is better than before.

File: src/main/java/formflow/library/ScreenController.java [hunk 2]
```diff
@@ -196,8 +260,8 @@ ModelAndView postScreen(
     // Address validation
     handleAddressValidation(submission, formSubmission);
 
-    // handle submit actions, if requested
-    if (request.getRequestURI().toLowerCase().contains("submit")) {
+    // handle marking the record as submitted, if necessary
```
Comment (spokenbird) [inline src/main/java/formflow/library/ScreenController.java:263]: Maybe do this as the last thing before right before save after the data merge from formData? Thoughts?

**Review by vrajmohan — APPROVED** [review]
LGTM

File: src/main/java/formflow/library/ScreenController.java (hunk 2 above)
Comment (bseeger) [reply src/main/java/formflow/library/ScreenController.java:271]: I moved this down past the merge but before the action. I think it's still good to have this before the action.

**Pull request merged**